### PR TITLE
Task 2 ready to review

### PR DESCRIPTION
**Pull request by HazRem**
Sorry for the big delay. Like i mentioned in the last commit message, I will explain myself in the Telegram chat as to why i took such a long time to send those Tasks

### CODE REVIEW

File: Homework/Task2_1/Point.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        Y = y;
+        Z = z;
+        Mass = mass;
+    }
+
+    public bool IsZero()
+    {
+        return X == 0 && Y == 0 && Z == 0;
+    }
+
```
Comment (IgorKheidorov) [inline Homework/Task2_1/Point.cs:42]: Check other for null, return -1 in this case

File: Homework/Task2_2(1)/MatrixClasses.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                {
+                    sb.Append(i == j ? diagonalElements[i].ToString() : "0");
+                    sb.Append(" ");
+                }
+                sb.AppendLine();
+            }
+            return sb.ToString();
+        }
+    }
+
```
Comment (IgorKheidorov) [inline Homework/Task2_2(1)/MatrixClasses.cs:88]: Move to a separate file

File: Homework/Task2_2(1)/MatrixClasses.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            return false;
+        }
+
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < Size; i++)
+            {
+                for (int j = 0; j < Size; j++)
+                {
```
Comment (IgorKheidorov) [inline Homework/Task2_2(1)/MatrixClasses.cs:79]: Use this[i, j] for consistency

File: Homework/Task2_2(1)/MatrixClasses.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                Size = elements.Length;
+                diagonalElements = new int[Size];
+                Array.Copy(elements, diagonalElements, Size);
+            }
+        }
+
+        public int this[int i, int j]
+        {
+            get
+            {
```
Comment (IgorKheidorov) [inline Homework/Task2_2(1)/MatrixClasses.cs:32]: Move index validation to a separate private method, and use t in lines 37 and 43

File: Homework/Task2_2(1)/MatrixClasses.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
+﻿using System;
```
Comment (IgorKheidorov) [inline Homework/Task2_2(1)/MatrixClasses.cs]: Remove unused usings

File: Homework/Task2_3/Entities.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2_3
+{
```
Comment (IgorKheidorov) [inline Homework/Task2_3/Entities.cs]: Move to classes to files

File: Homework/Task2_3/Training.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2_3
+{
+    public class Training : ICloneable
+    {
```
Comment (IgorKheidorov) [inline Homework/Task2_3/Training.cs:10]: array has to be used here -> TrainingContent[] contents

File: Homework/Task2_3/Training.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+                    return false;
+            }
+            return contents.Count > 0;
+        }
+
+        public object Clone()
+        {
+            Training clonedTraining = new Training();
+            foreach (var content in contents)
+            {
```
Comment (IgorKheidorov) [inline Homework/Task2_3/Training.cs:45]: implement  TrainingContent Clone() {}  for Lecture and PracticalLesson, and simplify
clonedTraining.Add(content.Clone())

**Review by IgorKheidorov — APPROVED** [review]
Please fix comments and merge.

File: Homework/Task2_1/Point.cs (hunk 1 above)
Comment (HazRem) [reply Homework/Task2_1/Point.cs:42]: Added line checking whether other is null

File: Homework/Task2_2(1)/MatrixClasses.cs (hunk 5 above)
Comment (HazRem) [reply Homework/Task2_2(1)/MatrixClasses.cs]: Removed

File: Homework/Task2_2(1)/MatrixClasses.cs (hunk 4 above)
Comment (HazRem) [reply Homework/Task2_2(1)/MatrixClasses.cs:32]: Added IsIndexValid method

File: Homework/Task2_2(1)/MatrixClasses.cs (hunk 3 above)
Comment (HazRem) [reply Homework/Task2_2(1)/MatrixClasses.cs:79]: Fixed

File: Homework/Task2_3/Entities.cs (hunk 6 above)
Comment (HazRem) [reply Homework/Task2_3/Entities.cs]: Added Lecture, Training and Training Content class files, renamed Entities into PracticalLesson

File: Homework/Task2_3/Training.cs (hunk 8 above)
Comment (HazRem) [reply Homework/Task2_3/Training.cs:45]: Implemented

File: Homework/Task2_3/Training.cs (hunk 7 above)
Comment (HazRem) [reply Homework/Task2_3/Training.cs:10]: Changed the class to use an array instead of list

File: Homework/Task2_2(1)/MatrixClasses.cs (hunk 2 above)
Comment (HazRem) [reply Homework/Task2_2(1)/MatrixClasses.cs:88]: Moved to DiagonalMatrixExtensions

**Pull request merged**